domain/modelagent/service/service.go

```diff
@@ -10,12 +10,16 @@ import (
 	"github.com/juju/version/v2"
 
 	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/user"
 )
 
 // State provides the state methods needed by the modelagent service.
 type State interface {
 	// GetModelAgentVersion returns the agent version for the specified model.
 	GetModelAgentVersion(ctx context.Context, modelID model.UUID) (version.Number, error)
+	// AgentVersionForModelName returns the agent version for the model with the
+	// given name and owner.
+	AgentVersionForModelName(ctx context.Context, user user.Name, modelName string) (version.Number, error)
```

Comment:
This method, though not exposed in the service, invites folks to use it when they should be using GetModelAgentVersion instead. 

Why isn't the controller model UUID available when the ControllerAgentVersion() service method would be called?

At the very least it needs big warning signs about its use

Reply:
I wrote this a while ago, now there is a `controller` table that stores the controller model ID, so I could probably get it from there instead.

However out of curiosity, what exactly is your concern with retrieving an agent version by model name? It seems like a perfectly valid operation to me. The same way you can retrieve a model by UUID or by name. The same way you can retrieve a user by UUID or by username. Etc.

Reply:
The goal is to leave names in the api server level and use UUIDs once something is created. Referring to items by name is a left over from tags which we're slowing removing. As well, there can be multiple models with the same name on a controller where the owner is different. Controller is a special case (I believe), where there can be only one.